src/ast/Visitor.h

```diff
@@ -59,10 +58,6 @@
 #include <vector>
 
 namespace souffle {
-class AstConstant;
-class AstFunctor;
-class AstTerm;
-class AstType;
```

Comment:
These are all used. The file will still compile due to transitive includes, but we have been trying to move away from unreliable dependencies. In this particular file other things are likely to break if those mistakes are made, but consistency and readability are easier than a list of exceptions.

Reply:
I don't like forward defs - we need to start including these classes but will keep it as is.